src/route-sync/config/config.go

```diff
@@ -1,47 +1,148 @@
 package config
 
 import (
-	"encoding/json"
+	"fmt"
+	"io/ioutil"
 
+	yaml "gopkg.in/yaml.v2"
+
+	"code.cloudfoundry.org/multierror"
 	cfConfig "code.cloudfoundry.org/route-registrar/config"
-	"errors"
-	"github.com/kelseyhightower/envconfig"
 	uaaconfig "code.cloudfoundry.org/uaa-go-client/config"
 )
 
-// Config contains settings for the route-sync with envconfig annotations
+// ConfigSchema defines the file schema for the YAML configuration of route-sync
+type ConfigSchema struct {
```

Comment:
See PR description. I got this pattern from `route-registrar/config`. 

The good:

- We will not produce invalid `Config` objects. We refuse to convert if the ConfigScheme isn't valid. No awkward partially constructed object problems.
- Persistence is clearly separated from runtime config. These are two different concerns and makes testing them separately very straightforward. 

The bad:
- Duplication